src/ngx_http_lua_util.c

```diff
@@ -2192,137 +2192,70 @@ ngx_http_lua_escape_uri(u_char *dst, u_char *src, size_t size, ngx_uint_t type)
     return (uintptr_t) dst;
 }
 
+static int
+ngx_http_lua_util_hex2int(char xdigit)
+{
+    if (isdigit(xdigit)) {
+        return xdigit - '0';
+    } else {
```

Comment:
style: we can save the `else` branch here, since it will return in the `if` branch.

Reply:
```
    if (isdigit(xdigit)) {
        return xdigit - '0';
    }

    xdigit = tolower(xdigit);
```
this style can be better.